Automattic/wp-calypso
# Gutenlypso: Clean up Gutenberg layout file

#### Changes proposed in this Pull Request

Since Gutenberg's `edit-post` module isn't properly available to the outside world, we had to fall back to importing it manually, changing it where needed, and manually updating it every now and then being extra careful not to replace our own changes.

In a recent PR, #29758, I've noticed that the amount of customizations in `edit-post/components/layout/index.js` was growing out of control.
It's no surprise though, as it's the obvious place to add anything applying to the whole of Gutenlypso, exactly like it happens with the Calypso classic editor's `post-editor.jsx` file.

So I've considered moving all the `layout` customizations away from the `edit-post` folder and into a new `GutenlypsoLayout` override file, which will be the only (well, almost only) thing that we'll ever need to import in the original `layout` file, making it super simple to update in the future.

Note: this conflicts a lot with #29758

#### Testing instructions



* Create a new post with Gutenlypso (`/block-editor`)
* Make some changes and wait for the autosave to trigger.
* ⚠️ It should update the URL with the post ID.
* Publish the post. Make some changes and save it for a few times, until the Revisions link appear in the document sidebar.
* ⚠️ Make sure the Revision link opens the revisions dialog and work as expected.
